### ISSUE TITLE
Getting stuck on startup if conecting to last used session fails

### ISSUE THREAD

**Issue by AndisGrossteins** [HeidiSQL/HeidiSQL#774]
## Steps to reproduce this issue

1. Connect to a host via a SSH tunnel (might not matter);
2. Close the session (re-opening last used session has to be enabled);
3. Change credentials on the server (change SSH key and/or password);
4. Launch HeidiSQL
5. Observe PLink dialog asking for password and another popup dialog with message "PLink: Access denied" after that.

## Current behavior
HeidiSQL just loops between two dialogs but doesn't allow to cancel the operation.
Although, at some point HeidiSQL gives up waiting for plink.exe and displays a different error message - "PLink exited unexpectedly (...)". The connection failure is to be expected since user credentials have changed but that's not the issue here.

## Expected behavior
HeidiSQL should allow to exit the loop by displaying a "cancel connection attempt" button or similar and, after user clicks the button, open main window or session manager.

## Possible solution
See Expected behavior ⬆

## Environment

* **HeidiSQL version:** 
10.2.0.5721

* **Operating system:** 
Windows 10 v1903 (build 18362.418)

* **PLink.exe version:** _(if it matters)_
0.73.0 (from PuTTY 0.73 install which is in `%PATH%`)

**Comment by ansgarbecker** [comment]
plink 0.71 from July 2019 does not seem to cause this issue here. The output messages must have been changed in the newer plink builds.

**Comment by AndisGrossteins** [comment]
I don't think the issue is caused by plink since the interactions are handled by HeidiSQL UI.

**Comment by ansgarbecker** [comment]
Well I didn't want to blame plink, so let me rephrase that: the newer plink versions _trigger_ this issue in HeidiSQL. The issue itself lies in the HeidiSQL code.

**Comment by ansgarbecker** [comment]
@AndisGrossteins could you please try to reproduce that loop with the current HeidiSQL build? I am not able to get into some loop, anyway if I reconnect automatically or just connect by click.

**Comment by AndisGrossteins** [comment]
I can still reproduce the loop when starting a connection from session manager window. But, as I already wrote, it happens only when the connection credentials are not valid (server rejects the SSH key). The loop seems to repeat only three times so it might not be so bad.

[I made a screen recording of the ordeal](https://i.imgur.com/6ApXR8z.gif).

One more thing I noticed: It seems that the connection code is running on main thread when a failing connection (it timeouts) is started from the session manager window. UI becomes unresponsive and window title shows the dreaded "(Not responding)" message until a connection timeout is reached.

**Comment by ansgarbecker** [comment]
Ok, I just provided a wrong password, and get 1 message only then. But it seems I have to use a wrong SSH key to reproduce that.

**Comment by ansgarbecker** [comment]
So my just pushed change now exits the wait loop as soon as the server sends an "Access denied" error. In my case this just worked on two different servers, until my IP was banned due to too many login errors :)

Please update and report back if that works for you.
